syntax = "proto3";

package order;

service OrderService {
  rpc AddItem (AddItemRequest) returns (OrderResponse) {}
  rpc GetOrder (GetOrderRequest) returns (OrderResponse) {}
  rpc Checkout (CheckoutRequest) returns (OrderResponse) {}
  rpc GetCart (GetCartRequest) returns (GetCartResponse) {} 
  rpc GetOrders (GetOrdersRequest) returns (GetOrdersResponse) {}
}

message AddItemRequest {
  int32 user_id = 1;
  int32 product_id = 2;
  int32 qty = 3;
  float price = 4;
}

message GetOrderRequest {
  int32 user_id = 1; 
}

message CheckoutRequest {
  int32 user_id = 1;
}

message GetCartRequest {
  int32 user_id = 1;
}

message GetOrdersRequest {
  int32 user_id = 1;  
}

message OrderResponse {
  int32 id = 1;
  string items = 2;
  bool is_open = 3;
  int32 user_id = 4; 
  string date_added = 5;
  float amount = 6;
}

message GetCartResponse {
    string items = 1;
}

message GetOrdersResponse {
  repeated OrderResponse orders = 1; 
}

message OrderItem {
  float price = 1;
  int32 product = 2;
  int32 quantity = 3;  
}